pytorch_lightning/trainer/progress.py

```diff
@@ -141,22 +141,22 @@ def scheduler_steps(self) -> int:
 
 
 @dataclass
-class TrainingProgress(Progress):
```

Comment:
Why did you get rid of this class?

It is there to define a custom dataclass just for the `TrainingLoopProgress` epoch counter.

This is because there is no point to track optimization at the batch level

Reply:
I changed it as I see value in having a batch level tracking for optimizer. 

I am using batch level tracking to know how many optimizer have been call in the current batch.

Reply:
What do you mean? 1 batch equals 1 forward and 1 forward equals 0 or 1 steps. Am I missing something? 

Also what would be the distinction between total and current for batch optimization?
And would you want schedulers too?